clang/test/Driver/pch-fsycl-error.cpp

```diff
@@ -0,0 +1,14 @@
+// This test checks that an error is emitted when 
+// PCH(Pre-Compiled Header) file generation is forced in -fsycl mode.
+
+// RUN: touch %t.h
+
+// Linux
+// RUN: %clang -c -fsycl -x c++-header %t.h -###  %s 2> %t1.txt
+// RUN: FileCheck %s -input-file=%t1.txt
+// CHECK: Pre-compiled header generation is not supported with '-fsycl'
+
+// Windows
+// RUN: %clang_cl -c -fsycl -x c++-header %t.h -### -- %s 2>&1 \
```

Comment:
The MSVC compatible driver (clang-cl) also supports usage of `/Yc` for PCH creation please test for that too.